samples/Ark.ResourceWatcher/Ark.ResourceWatcher.Sample/Transform/CsvTransformService.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed under the MIT License. See LICENSE file for license information.
+using System.Globalization;
+using System.Text;
+
+using Ark.ResourceWatcher.Sample.Dto;
+
+using CsvHelper;
+using CsvHelper.Configuration;
+
+namespace Ark.ResourceWatcher.Sample.Transform;
+
+/// <summary>
+/// Transforms CSV byte content to SinkDto.
+/// </summary>
+public sealed class CsvTransformService
+{
+    private readonly string _sourceId;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CsvTransformService"/> class.
+    /// </summary>
+    /// <param name="sourceId">The source identifier for the output.</param>
+    public CsvTransformService(string sourceId)
+    {
+        _sourceId = sourceId;
+    }
+
+    /// <summary>
+    /// Transforms CSV byte content to SinkDto.
+    /// </summary>
+    /// <param name="input">The CSV data as a byte array.</param>
+    /// <returns>The transformed SinkDto.</returns>
+    public SinkDto Transform(byte[] input)
+    {
+        ArgumentNullException.ThrowIfNull(input);
+
+        var content = Encoding.UTF8.GetString(input);
+        
+        // Handle empty input
+        if (string.IsNullOrWhiteSpace(content))
+        {
+            return new SinkDto
+            {
+                SourceId = _sourceId,
+                Records = []
+            };
+        }
+        
+        using var reader = new StringReader(content);
+        var config = new CsvConfiguration(CultureInfo.InvariantCulture)
+        {
+            HasHeaderRecord = true,
+            TrimOptions = TrimOptions.Trim,
+            HeaderValidated = null, // Don't validate headers - allow extra columns
+            PrepareHeaderForMatch = args => args.Header.ToLowerInvariant() // Case-insensitive matching
+        };
+        
+        using var csv = new CsvReader(reader, config);
+
+        csv.Context.RegisterClassMap<SinkRecordMap>();
+
+        List<SinkRecord> records;
+        try
+        {
+            records = csv.GetRecords<SinkRecord>().ToList();
+        }
+        catch (CsvHelper.CsvHelperException ex) when (ex.InnerException is FormatException formatEx)
+        {
+            // Unwrap FormatException from CsvHelperException for invalid data types
+            throw formatEx;
+        }
+        catch (CsvHelper.CsvHelperException ex) when (ex.InnerException?.InnerException is FormatException formatEx2)
+        {
+            // Unwrap FormatException that's nested deeper (e.g., through TypeConverterException)
+            throw formatEx2;
+        }
+        catch (CsvHelper.CsvHelperException ex) when (ex.InnerException is CsvHelper.MissingFieldException)
+        {
+            // Handle missing required fields with a clear error message
+            var row = ex.Context?.Parser?.Row ?? 0;
+            throw new TransformException($"Invalid CSV format at line {row}: expected at least 3 columns", ex);
+        }
+        catch (CsvHelper.CsvHelperException ex)
```

Comment:
remove all other catches and simply this. This is overly complex only for 'testing' purpose.

move the complexity in the Test Assertion if needed by try matching over any of the inner exceptions instead of only on the root.